File: Assignment/GroupAssignment/frontend/src/types/dto/reservation.dto.ts
export enum ReservationStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  CONFIRMED = 'CONFIRMED',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
  NO_SHOW = 'NO_SHOW'
}

export interface ReservationDTO {
  reservationId: string;
  branchId: string;
  areaTableId?: string;
  startTime: string;
  customerName: string;
  customerPhone: string;
  customerEmail?: string;
  guestNumber: number;
  note?: string;
  status: ReservationStatus;
  arrivalTime?: string;
  completionTime?: string;
  rejectionReason?: string;
  createdAt: string;
  updatedAt: string;
  branchName?: string;
  branchAddress?: string;
  tableTag?: string;
  tableCapacity?: number;
  serviceDurationMinutes?: number;
  estimatedDurationMinutes?: number;
}

export interface CreateReservationRequest {
  branchId: string;
  areaTableId?: string;
  startTime: string;
  customerName: string;
  customerPhone: string;
  customerEmail?: string;
  guestNumber: number;
  note?: string;
  estimatedDurationMinutes?: number;
}

export interface RejectReservationRequest {
  reason?: string;
}

export interface ReservationAnalyticsDTO {
  totalReservations: number;
  pendingCount: number;
  approvedCount: number;
  confirmedCount: number;
  completedCount: number;
  cancelledCount: number;
  noShowCount: number;
  approvalRate: number;
  noShowRate: number;
  averageServiceDurationMinutes: number;
  reservationsByDate: Record<string, number>;
  reservationsByTimeSlot: Record<string, number>;
}

export interface ReservationFilterParams {
  statuses?: ReservationStatus[];
  startDate?: string;
  endDate?: string;
  search?: string;
}

export type TableAvailabilityStatus = 'AVAILABLE' | 'RISKY' | 'UNAVAILABLE';

export interface TableAvailabilityDTO {
  tableId: string;
  tableTag: string;
  capacity: number;
  status: TableAvailabilityStatus;
  reason?: string;
}

export interface GetAvailableTablesParams {
  branchId: string;
  time: string; // ISO datetime
  guests: number;
  duration?: number; // minutes
}

File: Assignment/GroupAssignment/backend/src/main/java/com/example/backend/dto/ReservationDTO.java
package com.example.backend.dto;


import com.example.backend.entities.ReservationStatus;
import lombok.Data;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class ReservationDTO {

    private UUID reservationId;

    private UUID branchId;

    private UUID areaTableId;

    private LocalDateTime startTime;

    private String customerName;

    private String customerPhone;

    private String customerEmail;

    private int guestNumber;

    private String note;

    private ReservationStatus status;

    private Instant arrivalTime;

    private Instant completionTime;

    private String rejectionReason;

    private Instant createdAt;

    private Instant updatedAt;

    private String branchName;

    private String branchAddress;

    private String tableTag;

    private Integer tableCapacity;

    private Long serviceDurationMinutes;

    private Integer estimatedDurationMinutes;
}